app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -219,18 +231,34 @@ open class HomeActivity : LocaleAwareAppCompatActivity() {
         themeManager.setActivityTheme(this)
         themeManager.applyStatusBarTheme(this)
     }
-
-    private fun setupToolbarAndNavigation() {
+    /**
+     * Returns the inflated [supportActionBar].
+     * Callers trying to get the [supportActionBar] from the Activity should call this.
+     * Name chosen to be more discoverable and avoid confusion with references to getToolbar
+     * We initially use a ViewStub for startup performance benefits, and
+     * after inflation, it becomes a Toolbar.
+     */
+    fun getSupportActionBarAndInflateIfNecessary(): ActionBar {
         // Add ids to this that we don't want to have a toolbar back button
-        val appBarConfiguration = AppBarConfiguration.Builder().build()
-        val navigationToolbar = findViewById<Toolbar>(R.id.navigationToolbar)
-        setSupportActionBar(navigationToolbar)
-        NavigationUI.setupWithNavController(navigationToolbar, navHost.navController, appBarConfiguration)
-        navigationToolbar.setNavigationOnClickListener {
-            onBackPressed()
-        }
+        if (!isToolbarInflated) {
+            val navigationToolbar = (navigationToolbarStub.inflate() as Toolbar).also {
+                this.navigationToolbar = it
```

Comment:
nit: we don't actually need this reference if we return `supportActionBar!!` instead.